File: Apps/Codaxy.Dextop.Showcase/Demos/Grids/GridWindow.js
Ext.ns('Showcase');

Ext.define('Showcase.demos.GridWindow', {
	extend: 'Dextop.Window',
	width: 500,
	height: 300,

	title: 'Grid Window',
	requires: 'Ext.grid.GridPanel',

	initComponent: function () {
		var store = this.remote.createStore('model', {
			autoLoad: true
		});

		var columns = this.remote.createGridColumns('model');

		var grid = Ext.create('Ext.grid.GridPanel', {
			border: false,
			store: store,
			columns: columns
		});

		Ext.apply(this, {
			layout: 'fit',
			items: grid
		});

		this.callParent(arguments);
	}
});

File: Apps/Codaxy.Dextop.Showcase/Demos/Grids/GridWindow.cs
using System;
using Codaxy.Dextop.Data;

namespace Codaxy.Dextop.Showcase.Demos.Grids
{
    [Demo("GridWindow",
        Title = "Basic Grid",
        Description = "Dextop helps you with the Grid.",
        Path = "~/Demos/Grids"
    )]
    [LevelBeginner]
    [TopicDextopGrid]
    [CategoryFeature]
    public class GridWindow : DextopWindow
    {
        public override void InitRemotable(DextopRemote remote, DextopConfig config)
        {
            base.InitRemotable(remote, config);
			Remote.AddStore("model", Read);
        }

		GridModel[] Read(DextopReadFilter filter)
		{
			return new[] {
                new GridModel { Id = 1, Name = "Bill", Age = 30, Height=180 },
                new GridModel { Id = 2, Name = "Bob", Age = 26, Height=175 }               
            };
		}       

        [DextopModel]
        [DextopGrid]
        class GridModel
        {
            [DextopModelId]
            public int Id { get; set; } 
            
            [DextopGridColumn(flex=1)]
            public String Name { get; set; }
            
            [DextopGridColumn(width = 70)]            
            public int Age { get; set; }

            [DextopGridColumn(width = 70, text="Height (cm)", tooltip="Height in centimeters", sortable = false)]
            public int Height { get; set; }
        }
    }
}